### PR TITLE
Cleanup ReadAsync calls, make IISHttpContext private. 

### PR DESCRIPTION
**Pull request by jkotalik**
Offline feedback for 5e74b36c53038116c6a789a1e2a9b084f59dc377 that @davidfowl had.

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs [hunk 1]
```diff
@@ -39,7 +39,18 @@ public override int Read(byte[] buffer, int offset, int count)
 
         public override unsafe Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
-            return _httpContext.ReadAsync(buffer, offset, count, cancellationToken);
+            if (buffer == null)
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs]: Can we just let the Memory constructor handle this?

File: src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs [hunk 2]
```diff
@@ -39,7 +39,18 @@ public override int Read(byte[] buffer, int offset, int count)
...
+            {
+                throw new ArgumentNullException(nameof(buffer));
+            }
+            if (count == 0)
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs]: This shouldn't throw.

File: src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs (hunk 1 above)
Comment (jkotalik) [reply src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs]: Doesn't throw with a null buffer.

Comment (davidfowl) [reply src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs]: Then maybe it shouldn't.

File: src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs (hunk 2 above)
Comment (Tratcher) [reply src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs]: Stream.Read/Async should not return 0 until the end of the body. You need a non-zero buffer to do that.

**Pull request merged**